### DISCUSSION TITLE
Trouble opening sas file with UNC path

### DISCUSSION THREAD

**Discussion by lucielerolland** [Roche/pyreadstat#135] [Q&A] [answered]
Hi all,  

I'm having trouble using read_sas7bdat and I was wondering if anyone had ever faced something similar. 

I'm on a Windows 10 64 bits platform using pyreadstat version 1.1.2. I'm trying to read a SAS file located at a UNC path (think : \\server_name\path\to\my\file.sas7dbat). 

The path exists, meaning that :
`my_unc_path = r'\\server_name\path\to\my\file.sas7dbat'`
`print(os.path.exists(my_unc_path))`
returns True. I'm also able to pd.read_sas the file in question. 

However, when I try to run :
`file, meta = read_sas7bdat(my_unc_path)`

I get the following : "pyreadstat._readstat_parser.PyreadstatError: File \\\\server_name\path\to\my\file.sas7dbat does not exist !"

I'm able to read some regular, C:\Users\MyName\path\to\my\file.sas7dbat files, so I'm inclined to think it's probably a UNC path issue. 

Any idea what's going on here and how I might manage to read my file ? 

Thanks !

Best, 

Lucie

**Comment by ofajardo** [comment, marked as answer]
hmm, maybe try using forward slashes? https://stackoverflow.com/questions/7169845/using-python-how-can-i-access-a-shared-folder-on-windows-network

otherwise, does this sequence of commands reproduce the issue?

```
my_unc_path = r'\\server_name\path\to\my\file.sas7dbat'
filename_bytes = os.fsencode(my_unc_path)
filename_bytes = os.path.expanduser(filename_bytes)
os.path.isfile(filename_bytes)
```

the last line should rise an error, you could print what is filename_bytes to see why os.path.isfile thinks it is not a file.

**Reply by ofajardo** [reply]
in my case this sequence of commands do work correctly, how is it for you?
For me also reading a sas7bdat from an UNC path works as well. 
IF this sequence of commands is not working I may have something to do with the system encoding of your machine ... what is it in your case? is the encoding of the remote server the same?

**Reply by ofajardo** [reply]
oh, by the way, please make sure you are using the latest version of pyreadstat, there were some issues with opening files in older versions.

**Reply by ofajardo** [reply]
does your UNC path contains non ascii characters? does reading from a path with only ascii characters work?

another thing you could try is to covert the string path to bytes yourself (not using fsencode) making sure the conversion is correct and pass that to pyreadstat, in that case it should work

**Comment by lucielerolland** [comment]
Hi, thanks a lot for trying all this ! 
The offender is in fact the connection between the server and my machine. When asked about this, the server's owner advised me to copy the files and work locally.
Cheers and thanks for the work on this module ! It's very much appreciated.
